libraries/SPI/src/SPI.cpp

```diff
@@ -153,6 +154,7 @@ void SPIClass::endTransaction()
     if(_inTransaction){
         _inTransaction = false;
         spiEndTransaction(_spi);
+	spiUnlock(_spi);
```

Comment:
Possible bug.  spiEndTransaction() is the same as calling spiUnlock(), so you are calling unlock twice.